src/workflows-app/components/RecordsTable.ts

```diff
@@ -0,0 +1,188 @@
+import { h, span } from 'react-hyperscript-helpers';
+import { AutoSizer } from 'react-virtualized';
+import { Checkbox } from 'src/components/common';
+import { GridTable, HeaderCell, Resizable, Sortable, TextCell } from 'src/components/table';
+import {
+  RecordAttributes,
+  RecordResponse,
+  RecordTypeSchema,
+} from 'src/libs/ajax/data-table-providers/WdsDataTableProvider';
+import colors from 'src/libs/colors';
+import { parseAttributeName } from 'src/workflows-app/utils/submission-utils';
+
+const recordMap = (records: RecordResponse[]) => {
+  return _.fromPairs(_.map((e) => [e.id, e], records));
+};
+
+type RecordsTableProps = {
+  records: RecordResponse[];
+  selectedRecords: Record<string, RecordResponse>;
+  setSelectedRecords: Dispatch<SetStateAction<Record<string, RecordResponse>>>;
+  selectedDataTable: RecordTypeSchema;
+};
+
+const RecordsTable = ({ records, selectedRecords, setSelectedRecords, selectedDataTable }: RecordsTableProps) => {
+  const [dataTableColumnWidths, setDataTableColumnWidths] = useState({});
+  const dataTableRef = useRef<{ recomputeColumnSizes: () => void }>(null);
+  const [recordsTableSort, setRecordsTableSort] = useState<{ field: string; direction: 'asc' | 'desc' }>({
+    field: 'id',
+    direction: 'asc',
+  });
+
+  useEffect(() => {
+    dataTableRef.current?.recomputeColumnSizes();
+  }, [dataTableColumnWidths, records]);
+
+  const selectAll = () => {
+    setSelectedRecords(_.assign(selectedRecords, recordMap(records)));
+  };
+
+  const deselectAll = () => {
+    setSelectedRecords({});
+  };
+
+  const allSelected = () => {
+    const recordIds = _.map('id', records);
+    const selectedIds = _.keys(selectedRecords);
+    return records.length && _.every((k) => _.includes(k, selectedIds), recordIds);
+  };
+
+  const resizeColumn = (currentWidth, delta, columnKey) => {
+    setDataTableColumnWidths(_.set(columnKey, currentWidth + delta));
+  };
+
+  const withDataTableNamePrefix = (columnName) => `${selectedDataTable.name}/${columnName}`;
+
+  const recordsTableData = _.orderBy<RecordResponse & RecordAttributes>(
+    [
+      ({ [recordsTableSort.field]: field }) =>
+        Number.isNaN((field as any) * 1) ? _.lowerCase(field as string) : (field as any) * 1, // converts field to int, float, or NaN (if field is a string)
+    ],
+    [recordsTableSort.direction],
+    records.map((row) => ({
+      ...row,
+      ...row.attributes,
+    }))
+  );
+
+  const renderCellData = (data) => {
+    if (_.isObject(data) || _.isBoolean(data)) {
+      return JSON.stringify(data);
+    }
+    return data;
+  };
+
+  return h(AutoSizer, { disableHeight: true }, [
+    ({ width }) =>
+      h(GridTable, {
+        'aria-label': `${selectedDataTable.name} data table`,
+        ref: dataTableRef,
+        // @ts-ignore
+        sort: recordsTableSort,
+        width,
+        height: (1 + recordsTableData.length) * 48,
```

Comment:
Main functional change is here - we are not using AutoSizer to determine the height and are instead calculating it based on the number of rows * default row height